File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
       HashSet<Integer> set=new HashSet<>();
       int max=nums[0];
       for(int i=0;i<nums.length;i++)
       {
            max=Math.max(max,nums[i]);
            set.add(nums[i]);
       }
        for(int i=0;i<=max;i++)
        {
            if(!set.contains(i))
            {
                return i;
            }
        }
        return max+1;
    }
}

File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        int l=0,h=n-1;
        if(l!=nums[l]) return 0;
        if(h==nums[h]) return h+1;
        while(l+1<h){
            int mid=(l+h)/2;
            if(nums[mid]==mid) l=mid;
            else h=mid;
        }
        return l+1;
    }
};